core/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java

```diff
@@ -363,6 +364,7 @@ public void testSegments() throws Exception {
             assertThat(segments.get(1).isCompound(), equalTo(false));
 
             engine.config().setCompoundOnFlush(true);
+            engine.onSettingsChanged();
```

Comment:
Because the test previously relied on `refresh` to make this settings change live, which I think is a poor API.